game-core/src/main/java/games/strategy/triplea/ui/screen/drawable/TerritoryDrawable.java

```diff
@@ -9,23 +9,22 @@
 
 import games.strategy.engine.data.Territory;
 import games.strategy.triplea.ui.mapdata.MapData;
+import games.strategy.ui.Util;
 
 public abstract class TerritoryDrawable {
   protected static void draw(final Rectangle bounds, final Graphics2D graphics, final MapData mapData,
       final Territory territory, final Paint territoryPaint) {
     final List<Polygon> polys = mapData.getPolygons(territory);
-    for (Polygon polygon : polys) {
-      // if we dont have to draw, dont
+    for (final Polygon polygon : polys) {
       if (!polygon.intersects(bounds) && !polygon.contains(bounds)) {
         continue;
       }
-      // use a copy since we will move the polygon
-      polygon = new Polygon(polygon.xpoints, polygon.ypoints, polygon.npoints);
-      polygon.translate(-bounds.x, -bounds.y);
+
+      final Polygon translatedPolygon = Util.translatePolygon(polygon, -bounds.x, -bounds.y);
       graphics.setPaint(territoryPaint);
-      graphics.fillPolygon(polygon);
+      graphics.fillPolygon(translatedPolygon);
       graphics.setColor(Color.BLACK);
```

Comment:
Could not hoist `setPaint()` and `setColor()` outside of the loop because these two methods are interdependent (i.e. calling `setPaint()` changes the stroke color, so it must be reset to black before drawing the polygon).